### PR TITLE
DATACOUCH-227 - Keep isOrderRetained view param, bump to SDK 2.2.8

### PR DESCRIPTION
**Pull request by simonbasle**
@daschl can you do a quick review? :bow:

### CODE REVIEW

File: src/main/java/org/springframework/data/couchbase/core/CouchbaseTemplate.java [hunk 1]
```diff
@@ -304,9 +304,12 @@ public RawJsonDocument doInBucket() {
   public <T> List<T> findByView(ViewQuery query, final Class<T> entityClass) {
     //we'll always need to get documents, as a RawJsonDocument, so we should force that target class
     //so that the caller doesn't set a bad target class unintentionally, pre-loading with a bad type.
-    //TODO DATACOUCH-227 reproduce retainOrder parameter
     if (!query.isIncludeDocs() || !query.includeDocsTarget().equals(RawJsonDocument.class)) {
-      query.includeDocs(RawJsonDocument.class);
+      if (!query.isOrderRetained()) {
```
Comment (daschl) [inline src/main/java/org/springframework/data/couchbase/core/CouchbaseTemplate.java]: can you turn the if order around? having ! and an else block is confusing :)

File: src/integration/java/org/springframework/data/couchbase/core/CouchbaseTemplateTests.java [hunk 2]
```diff
@@ -499,6 +491,22 @@ public void expiryWhenTouchOnReadDocument() throws InterruptedException {
 		assertNull(template.findById(id, DocumentWithTouchOnRead.class));
 	}
 
+	@Test
+	public void shouldRetainOrderWhenQueryingViewOrdered() {
+		ViewQuery q = ViewQuery.from("test_beers", "by_name");
+		q.descending()
+				.includeDocsOrdered(true);
```
Comment (daschl) [inline src/integration/java/org/springframework/data/couchbase/core/CouchbaseTemplateTests.java]: formatting

**Comment by daschl** [comment]
👍

**Comment by simonbasle** [comment]
cherry-picked as d6b63c3 and backported to `2.1.x` as 99bbc42

**Pull request closed without merging**